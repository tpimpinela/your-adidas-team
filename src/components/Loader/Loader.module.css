.loader-container {
  width: 100%;
  height: 100%;
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  flex-direction: column;
  gap: 25px;
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.9);
  z-index: 2;

  .loader {
    border: 16px solid var(--light-grey-color);
    border-top-color: var(--accent-color);
    width: 60px;
    height: 60px;
    animation: spin 2s linear infinite;
    border-radius: 50%;
  }
}

@keyframes spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
