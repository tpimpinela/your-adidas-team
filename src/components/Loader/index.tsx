import styles from "./Loader.module.css";

const Loader = () => (
  <div className={styles["loader-container"]}>
    <div className={styles.loader}></div>
    <span>Loading...</span>
  </div>
);

export default Loader;
