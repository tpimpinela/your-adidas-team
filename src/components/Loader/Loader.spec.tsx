import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Loader from ".";

describe("<Loader />", () => {
  it("should match the snapshot", () => {
    const { container } = render(<Loader />);
    expect(container).toMatchSnapshot();
  });
});
