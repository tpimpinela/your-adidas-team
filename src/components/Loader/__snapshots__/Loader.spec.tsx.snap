// Vitest Snapshot v1

exports[`<Loader /> > should match the snapshot 1`] = `
<div>
  <div
    class="_loader-container_bf668d"
  >
    <div
      class="_loader_bf668d"
    />
    <span>
      Loading...
    </span>
  </div>
</div>
`;
